packages/types/src/interfaces/metadata/definitions.ts

```diff
@@ -78,6 +78,9 @@ export default {
         V14: 'MetadataV14',
         V15: 'MetadataV15'
       }
-    }
+    },
+
+    // This is the CheckMetadataHash SignedExtension type.
+    MetadataHash: 'Option<Vec<u8>>'
```

Comment:
So the `mode` is an enum/u8. And the additional signed is a special type that encodes to nothing if `mode == 0`, otherwise it only passes a `[u8; 32]`.